geomstats/information_geometry/multivariate_normal.py

```diff
@@ -0,0 +1,348 @@
+        n = self.n
+
+        def pdf(x):
+            """Generate parameterized function for normal pdf.
+
+            Parameters
+            ----------
+            x : array-like, shape=[n_samples, n]
+                Points at which to compute the probability
+                density function.
+            """
+            x = gs.to_ndarray(x, to_ndim=2, axis=0)
+            x = x[None, :, :]
+            det_cov = gs.squeeze(gs.prod(diagonal, axis=-1))
+            tmp_0 = 1 / gs.sqrt(gs.power((2 * gs.pi), n) * det_cov)
+            tmp_1 = gs.exp(-0.5 * gs.sum(((x - location) ** 2) / diagonal, axis=-1))
+            while tmp_0.ndim < tmp_1.ndim:
+                tmp_0 = tmp_0[..., None]
+            pdf_at_x = tmp_0 * tmp_1
+            pdf_at_x = gs.squeeze(pdf_at_x)
+            return pdf_at_x
+
+        return pdf
+
+
+class MultivariateDiagonalNormalMetric(RiemannianMetric):
+    """Class for the Fisher information metric of diagonal normal distributions."""
+
+    def __init__(self, n):
+        self.n = n
+        dim = int(2 * n)
+        super().__init__(dim=dim)
+        self.univariate_normal_metric = NormalMetric()
+
+    def _unstack_location_diagonal(self, point):
+        """Extract location and diagonal of the covariance matrix from a given point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 2*n]
+            Input point from which locations and diagonals are extracted.
+
+        Returns
+        -------
+        location : array-like, shape=[..., n]
+            Locations from the input point.
+        diagonal : array-like, shape=[..., n]
+            Diagonals of covariance matrices from the input point.
+        """
+        location = point[..., : self.n]
+        diagonal = point[..., self.n :]
+        return location, diagonal
+
+    def _stack_location_diagonal(self, location, diagonal):
+        """Set location and diagonal of the covariance matrix into a point.
+
+        Parameters
+        ----------
+        location : array-like, shape=[..., n]
+            Locations to stack.
+        diagonal : array-like, shape=[..., n]
+            Diagonals of covariance matrices from the input point.
+
+        Returns
+        -------
+        point : array-like, shape=[..., 2*n]
+            Point with locations and diagonals covariance matrices.
+        """
+        point = gs.concatenate([location, diagonal], axis=-1)
+        return point
+
+    def _stacked_location_diagonal_to_1d_pairs(self, point, apply_sqrt=False):
+        """Create pairs of 1d locations and stds from nd counterparts.
+
+        Parameters
+        ----------
+        point: array-like, shape=[..., 2*n]
+            Point.
+        apply_sqrt: bool
+            Determine if a square root is applied to the diagonals.
+
+        Returns
+        -------
+        pairs : array-like, shape=[..., n, 2]
+            Pairs of locations and standard deviations.
```

Comment:
By default, pairs of locations and variances ? When apply_sqrt is True, then standard deviations instead ?

Reply:
Not necessarily since this function can also be applied to tangent vectors.